File: houduan/model/quiz.go
package model

import "time"

// TableName 显式指定表名:GORM 默认复数化会把 QuizQuestion 映射为 quiz_questions
// 恰好一致,这里仍显式固定,与表注释维护的表名保持一致,避免依赖默认行为。
func (QuizQuestion) TableName() string { return "quiz_questions" }

// QuizQuestion 蜀文化知识闯关题目(景点答题得徽章,三语)。
// 题目由 cmd/quiz-gen 批量生成,严格基于 scenic_spots 库内 desc/culture_note 出题,
// 解析引用库内资料原文要点,保证可溯源;答题记录存前端 localStorage,不做账号绑定。
type QuizQuestion struct {
	ID uint `gorm:"primaryKey;comment:题目ID" json:"id"`
	// SpotID 所属景点:答题入口挂在景点详情页,按景点抽题。
	SpotID     uint   `gorm:"index;comment:景点ID(scenic_spots.id)" json:"spot_id"`
	QuestionZH string `gorm:"type:varchar(255);comment:题干中文" json:"question_zh"`
	QuestionEN string `gorm:"type:text;comment:题干英文(LLM翻译,参考值)" json:"question_en"`
	QuestionJA string `gorm:"type:text;comment:题干日文(LLM翻译,参考值)" json:"question_ja"`
	// Options* 三个选项的 JSON 数组字符串,如 ["武侯祠","望江楼","青羊宫"],正确项下标为 AnswerIdx。
	OptionsZH string `gorm:"type:text;comment:中文选项JSON数组(3项)" json:"options_zh"`
	OptionsEN string `gorm:"type:text;comment:英文选项JSON数组(3项)" json:"options_en"`
	OptionsJA string `gorm:"type:text;comment:日文选项JSON数组(3项)" json:"options_ja"`
	// AnswerIdx 正确选项下标(0-2),校验答案在后端完成,前端不接触明文正确项。
	AnswerIdx int `gorm:"comment:正确选项下标(0-2)" json:"answer_idx"`
	// Explain* 答案解析(三语),内容引用库内资料原文要点,保证可溯源。
	ExplainZH string    `gorm:"type:text;comment:中文解析(引用库内资料要点)" json:"explain_zh"`
	ExplainEN string    `gorm:"type:text;comment:英文解析(LLM翻译,参考值)" json:"explain_en"`
	ExplainJA string    `gorm:"type:text;comment:日文解析(LLM翻译,参考值)" json:"explain_ja"`
	CreatedAt time.Time `gorm:"comment:创建时间" json:"created_at"`
}

File: wennv/wenlv/src/api/quiz.ts
/** 蜀文化知识闯关接口封装(公开接口,无需登录) */
import { get } from './request'

/** 单道闯关题目:三语题干与选项(JSON 数组字符串),正确项由后端判分不回传 */
export interface QuizQuestionItem {
  id: number
  spot_id: number
  question_zh: string
  /** 英/日文题干(LLM 生成,可能为空,展示回落中文) */
  question_en?: string
  question_ja?: string
  /** 三语选项 JSON 数组字符串,如 ["武侯祠","望江楼","青羊宫"] */
  options_zh: string
  options_en?: string
  options_ja?: string
  /** 中文解析(引用库内资料要点) */
  explain_zh: string
  explain_en?: string
  explain_ja?: string
}

/** 答案校验结果:判分在后端完成,answer_idx 为正确项下标 */
export interface QuizCheckResult {
  question_id: number
  correct: boolean
  answer_idx: number
  explain_zh: string
  explain_en?: string
  explain_ja?: string
}

/** 按景点随机抽题:题目不足时后端返回全部,无题返回空数组(前端隐藏答题卡) */
export function listQuizQuestions(spotId: number, count = 5): Promise<QuizQuestionItem[]> {
  return get<QuizQuestionItem[]>('/quiz', { params: { spot_id: spotId, count } })
}

/** 校验答案:answer 为所选选项下标 */
export function checkQuizAnswer(questionId: number, answer: number): Promise<QuizCheckResult> {
  return get<QuizCheckResult>('/quiz/check', {
    params: { question_id: questionId, answer },
  })
}
